httpx/models.py

```diff
@@ -209,7 +209,18 @@ def __str__(self) -> str:
 
     def __repr__(self) -> str:
         class_name = self.__class__.__name__
-        url_str = str(self)
+        if self._uri_reference.userinfo:
+            url_str = self._uri_reference.copy_with(
```

Comment:
Can this be simplified to `self._uri_reference.copy_with(userinfo=f"{self.username}:[secure]").unsplit()`?

Reply:
as far as I understand, we just copy `scheme, authority, path, query, fragment` ([link](https://github.com/python-hyper/rfc3986/blob/2982509402519575282845adf1f39dc0195369e0/src/rfc3986/_mixin.py#L321))

Reply:
What if `self.username` is `None`?

Reply:
`copy_with()` supports [a `userinfo=...` argument](https://rfc3986.readthedocs.io/en/latest/user/parsing.html?highlight=userinfo#some-examples).

So, I think this section of code can be...

```python
    if self._uri_reference.userinfo:
        self._uri_reference.copy_with(userinfo=f"{self.username}:[secure]").unsplit()
```

Note that self.username cannot be `None` if some userinfo value is set.

Reply:
Nevermind, totally forgot that it's `self.username` not `uri_reference.username`. Carry on! :)